src/Normalizer/ContentEntityNormalizer.php

```diff
@@ -453,6 +436,26 @@ private function denormalizeTranslation($translation, $entity_id, $entity_uuid,
             $target_entity_type_id = $settings['target_type'];
           }
 
+          // For user target type try to map users by username, if it can't be
+          // mapped by username use the ID from replication configuration object.
+          if ($target_entity_type_id === 'user') {
+            $users = [];
+            if (isset($item['username'])) {
+              $users = $this->entityManager->getStorage('user')
+                ->loadByProperties(['name' => $item['username']]);
+            }
+            $user = reset($users);
+            if ($user instanceof UserInterface && $id = $user->id()) {
+              $translation[$field_name][$delta] = ['target_id' => $id];
+            }
+            else {
+              $translation[$field_name][$delta] = ['target_id' => $this->usersMapping->getUid()];
```

Comment:
I think that we should pass in more details to the `usersMapping` service here, so that the service has more control. Maybe something like:

```php
$target_id = $this->usersMapping->mapReferenceField($entity, $field_name)
```

This way, it's completely up to the service to implement any logic, like looking for username or email, or falling back on config.